Sources/FilterViewController/FilterViewController.swift

```diff
@@ -4,161 +4,77 @@
-    var applyButtonHiddenConstraintConstant: CGFloat {
-        return applySelectionButton.height
-    }
+    // MARK: - Private properties
 
-    public var showsApplySelectionButton: Bool {
-        didSet {
-            guard showsApplySelectionButton != oldValue else {
-                return
-            }
-            view.layoutIfNeeded()
-            applySelectionButton.alpha = showsApplySelectionButton ? 0 : 1
-            applySelectionButton.isHidden = false
-            applySelectionButtonBottomConstraint?.constant = showsApplySelectionButton ? 0 : applyButtonHiddenConstraintConstant
-            UIView.animate(withDuration: 0.25, animations: { [applySelectionButton, showsApplySelectionButton] in
-                applySelectionButton.alpha = showsApplySelectionButton ? 1 : 0
-                self.view.layoutIfNeeded()
-            }) { [applySelectionButton, showsApplySelectionButton] _ in
-                applySelectionButton.isHidden = !showsApplySelectionButton
-            }
-            UIView.animate(withDuration: 0.25) {
-                self.view.layoutIfNeeded()
-            }
-            parentApplySelectionButtonOwner?.showsApplySelectionButton = showsApplySelectionButton
-        }
-    }
+    private lazy var applyButtonTopAnchorConstraint = applySelectionButton.topAnchor.constraint(equalTo: view.bottomAnchor)
 
-    public required init?(filterInfo: FilterInfoType, dataSource: FilterDataSource, selectionDataSource: FilterSelectionDataSource, navigator: FilterNavigator) {
-        guard let child = ChildViewController(filterInfo: filterInfo, dataSource: dataSource, selectionDataSource: selectionDataSource) else {
-            return nil
-        }
+    // MARK: - Setup
 
-        self.filterInfo = filterInfo
+    init(selectionDataSource: FilterSelectionDataSource, navigator: FilterNavigator?) {
         self.selectionDataSource = selectionDataSource
         self.navigator = navigator
-        showsApplySelectionButton = false
-        filterContainerViewController = child
         super.init(nibName: nil, bundle: nil)
-        child.filterSelectionDelegate = self
     }
 
-    public required init?(coder aDecoder: NSCoder) {
+    required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
     public override func viewDidLoad() {
         super.viewDidLoad()
+        view.backgroundColor = .milk
+        setupApplyButton()
+    }
 
-        let childViewController = filterContainerViewController.controller
+    public override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        view.bringSubviewToFront(applySelectionButton)
+    }
 
-        addChild(childViewController)
-        setup(with: childViewController.view)
-        childViewController.didMove(toParent: self)
+    public func showApplyButton(_ show: Bool, animated: Bool = true) {
+        applyButtonTopAnchorConstraint.isActive = !show
```

Comment:
If changing the constant if better than activating/deactivating than apple has done a terrible job!

That is weird, there shouldn't be anything to layout before the animation?

Reply:
You call 'view.layoutIfNeeded()' to make sure all other things are done before your change, so normally it does do anything, but there are instances where you get weird behavior without it. Don't remember what it was this time when that was added though

Reply:
Can't find anything about the activating/deactivating performance right now. Maybe it was some old guide or wwdc video... 🤔 

But does the button actually disappear when you deactivate that constraint if it was previously visible? I would guess you need to add a new constraint, otherwise the constraints aren't enough to fully define the location...

Reply:
I'll add the layoutIfNeeded just in case.

WWDC mention that it's better to activate/deactivate than deleting and creating new constraints.

I have one constraint which is greaterThanOrEqualTo, activating this constraint will actually hide the button ;)